schemas/stsci.edu/asdf/0.1.0/core/column.yaml

```diff
@@ -0,0 +1,47 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/column"
+tag: "tag:stsci.edu:asdf/0.1.0/core/column"
+
+title: >
+  A column in a table.
+
+description: |
+  Each column contains a name and an array of data, and an optional description
+  and unit.
+
+type: object
+properties:
+  name:
+    description: The name of the column
+    type: string
+    pattern: "[A-Za-z_][A-Za-z0-9_]*"
+
+  data:
+    description: |
+      The array data for the column.  The array must have a scalar
+      datatype (that is, it may not be a structured array).
```

Comment:
I'm not sure what to do about this.  This just came from the restriction/design of `astropy.table.Table`.  But that might be letting the horse design the cart, so maybe ASDF shouldn't explicitly disallow it.  I don't know how difficult it would be to get `astropy.table.Table` to support recarray columns.